File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/operations/CreditOperations.java
package org.nirvanasms.api.operations;

import java.util.HashMap;
import java.util.Map;

import org.nirvanasms.api.core.base.CreditBase;
import org.nirvanasms.api.core.entity.ApiUser;
import org.nirvanasms.api.core.entity.Credit;
import org.nirvanasms.api.core.utility.Constants;
import org.nirvanasms.api.core.utility.Enums.RequestMethod;
import org.nirvanasms.api.core.utility.HttpRequest;
import org.nirvanasms.api.core.utility.ProcessResult;

public class CreditOperations extends CreditBase {

	@Override
	public ProcessResult<Credit> query(ApiUser apiUser) {
		 Map<String, String> queryStrings = new HashMap<String, String>();
         queryStrings.put("username", apiUser.getUserName());
         queryStrings.put("password", apiUser.getApiPassword());
         HttpRequest request = new HttpRequest(Constants.CreditQueryUrl, queryStrings, RequestMethod.Get);
         String response = request.getResponse();
         ProcessResult<Credit> result = new ProcessResult<Credit>(response);
         if (result.isSuccess())
         {
             int amount = Integer.parseInt(response.substring(3));
             result.setData(new Credit(amount));
         }
         return result;
	}

}

File: NirvanaSMS.Api.CSharp/Operations/CreditOperations.cs
using NirvanaSMS.Api.Core.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NirvanaSMS.Api.Core.Entity;
using NirvanaSMS.Api.Core.Utility;

namespace NirvanaSMS.Api.Operations
{
    public class CreditOperations : CreditBase
    {
        public override ProcessResult<Credit> Query(ApiUser apiUser)
        {
            Dictionary<string, string> queryStrings = new Dictionary<string, string>();
            queryStrings.Add("username", apiUser.UserName);
            queryStrings.Add("password", apiUser.ApiPassword);
            HttpRequest request = new HttpRequest(Constants.CreditQueryUrl, queryStrings, Enums.RequestMethod.Get);
            string response = request.GetResponse();
            ProcessResult<Credit> result = new ProcessResult<Credit>(response);
            if (result.IsSuccess)
            {
                int amount = int.Parse(response.Substring(2));
                result.Data = new Credit(amount);
            }
            return result;
        }
    }
}
